android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java

```diff
@@ -199,20 +228,50 @@ public void shareViaInstagramStories(String filePath, String attributionUrl, fin
 
   @ReactMethod
   public void shareViaFacebook(String filePath, String shareUrl, final Promise promise) {
-    if (!shareUrl.isEmpty()) {
-      ShareLinkContent content = new ShareLinkContent.Builder()
-              .setContentUrl(Uri.parse(shareUrl))
-              .build();
-      ShareDialog.show(this.reactContext.getCurrentActivity(), content);
-      return;
-    }
 
-    String mimeType = this.getMimeType(filePath);
-    if (mimeType.startsWith("image")) {
-      this.shareImageViaFacebook(filePath, promise);
+    String applicationUri = "com.facebook.katana";
+
+    if (checkAppInstall(applicationUri)) {
+      try {
+        Intent shareIntent = new Intent(Intent.ACTION_SEND);
+        shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+        if (!shareUrl.isEmpty()) {
+            shareIntent.putExtra(Intent.EXTRA_TEXT, shareUrl);
+            shareIntent.setType("text/plain");
+        } else {
+            shareIntent.putExtra(Intent.EXTRA_STREAM, this.uriFromFilePath(filePath));
+            shareIntent.setType(this.getMimeType(filePath));
+        }
+
+        shareIntent.setPackage(applicationUri);
+        this.reactContext.getCurrentActivity().startActivity(shareIntent);
+
+        promise.resolve(null);
+      } catch (Exception ex) {
+        ex.printStackTrace();
+        shareContentViaFacebook(filePath, shareUrl, promise);
+      }
     } else {
-      this.shareVideoViaFacebook(filePath, promise);
+      shareContentViaFacebook(filePath, shareUrl, promise);
     }
+
+    // TODO: This code does not work due to error in facebook application. Check
+    // with Facebook app new versions.
+    // if (!shareUrl.isEmpty()) {
+    // ShareLinkContent content = new ShareLinkContent.Builder()
+    // .setContentUrl(Uri.parse(shareUrl))
+    // .build();
+    // ShareDialog.show(this.reactContext.getCurrentActivity(), content);
+    // return;
+    // }
+    //
+    // String mimeType = this.getMimeType(filePath);
+    // if (mimeType.startsWith("image")) {
+    // this.shareImageViaFacebook(filePath, promise);
+    // } else {
+    // this.shareVideoViaFacebook(filePath, promise);
+    // }
```

Comment:
Please remove these lines for clarity, instead of commenting them out. We can easily access this code using the git history and this commit's diff.

Reply:
Removed.